vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueType Kind;
+    UInt64 Value;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisValue, JsiValueData[] args);
+
+  [webhosthidden]
+  interface IJsiHostObject
+  {
+    JsiValueData GetProperty(IJsiRuntime runtime, UInt64 name);
+    void SetProperty(IJsiRuntime runtime, UInt64 name, JsiValueData value);
+    IVectorView<UInt64> GetPropertyNames(IJsiRuntime runtime);
+  };
+
+  [webhosthidden]
+  interface IJsiRuntime
```

Comment:
>IJsiRuntime [](start = 12, length = 11)

I'm sure I'm just missing something (or misunderstand the purpose of JSI), but through what set of methods would one get C++ values out of an IJsiRuntime? It seems all the methods are centered around the notion of some opaque UInt64 "tokens".

Reply:
In JSI most of the Runtime interface methods are protected. They are not intended to be used directly. They are only used to pass values through the ABI boundary. Developers are going to use types that wrap up calls to these methods. By avoiding passing 'true' objects we can avoid extra memory allocations.

---
In reply to: [451861888](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451861888) [](ancestors = 451861888)